app/notifications/process_notifications.py

```diff
@@ -161,37 +161,24 @@ def persist_notification(
     # if simulated create a Notification model to return but do not persist the Notification to the dB
     if not simulated:
         dao_create_notification(notification)
-        increment_daily_limit_cache(
-            service.id,
-            notification_type,
-            key_type,
-            international_sms=notification_type == SMS_TYPE and str(notification.phone_prefix) != UK_PREFIX,
-        )
+        increment_daily_limit_caches(service, notification, key_type)
 
     return notification
 
 
-def increment_daily_limit_cache(service_id, notification_type, key_type, international_sms=False):
+def increment_daily_limit_caches(service, notification, key_type):
     if key_type == KEY_TYPE_TEST or not current_app.config["REDIS_ENABLED"]:
         return
 
-    for notification_type_ in [None, notification_type]:
-        cache_key = redis.daily_limit_cache_key(service_id, notification_type=notification_type_)
-        if redis_store.get(cache_key) is None:
-            # if cache does not exist set the cache to 1 with an expiry of 24 hours,
-            # The cache should be set by the time we create the notification
-            # but in case it is this will make sure the expiry is set to 24 hours,
-            # where if we let the incr method create the cache it will be set a ttl.
-            redis_store.set(cache_key, 1, ex=86400)
-        else:
-            redis_store.incr(cache_key)
+    increment_daily_limit_cache(service.id)
```

Comment:
I think we haven't added this type of cache key before the change?

Reply:
We do. Because of this line we set caches for notification types of `None` and `notification_type`:
https://github.com/alphagov/notifications-api/blob/7a91fb85183457fa02fa6457c300e79c9babb856/app/notifications/process_notifications.py#L178

I don’t this existing way of doing this (with a 2-item loop) is very clear to read.

You can see it’s covered by the tests here:
https://github.com/alphagov/notifications-api/blob/7a91fb85183457fa02fa6457c300e79c9babb856/tests/app/notifications/test_process_notification.py#L288-L291

Reply:
Ahh, you are right 😮